### ISSUE TITLE
Intersection types doesn't work as expected

### ISSUE THREAD

**Issue by pleerock** [Microsoft/TypeScript#26634]
having following types:

```ts
export interface Category {
    id: number;
    name: string;
}

export interface Post {
    id: number;
    title?: string;
    categories?: Category[];
}

export type QueryOptions<T> = {
    [P in keyof T]?:
        T[P] extends Array<infer U> ? QueryOptions<U> :
        boolean
} & {
    $args?: { someOption: number }
}
```

If I do following:

```ts
const query: QueryOptions<Post> = {
    id: true,
    title: true,
    categories: {
        id2: true
    }
}
```

it gives me error:

<img width="303" alt="screenshot 2018-08-23 20 09 19" src="https://user-images.githubusercontent.com/1753397/44534046-4e504980-a710-11e8-9b0e-e4d0200e3f49.png">

However once I use my $args inside it doesn't give error on "id2" anymore

<img width="289" alt="screenshot 2018-08-23 20 10 13" src="https://user-images.githubusercontent.com/1753397/44534096-6cb64500-a710-11e8-81e3-df39d42bf566.png">

I expect it to have same error as in the first screenshot. 

p.s. sorry for the bad title, hope someone help to make it more concrete.

**Comment by andy-ms** [comment]
Duplicate of  #18075
